app/ui/legacy/src/main/java/com/fsck/k9/activity/setup/GoogleOAuthHelper.kt

```diff
@@ -0,0 +1,8 @@
+package com.fsck.k9.activity.setup
+
+object GoogleOAuthHelper {
+    fun isGoogle(hostname: String): Boolean {
+        return hostname.lowercase().endsWith("gmail.com") ||
+            hostname.lowercase().endsWith("googlemail.com")
```

Comment:
Please check for the exact host names. In cases like this, being too generic often comes back to bite you in the ass.

If we wanted to check for subdomains, the `endsWith()` argument would need to start with a dot because e.g. `not-gmail.com` will pass the current check.